app/audit/AuditService.scala

```diff
@@ -51,12 +51,21 @@ class AuditService @Inject() (auditConnector: AuditConnector)(implicit ec: Execu
     //todo Find out whether the bank account check needs to be in the event, if they get this far this should be a successful check
     //at this stage all optional values should be present
     val utr: SaUtr = submission.taxpayer.selfAssessment.utr
-    val name: String = submission.bankDetails.map(_.accountName.get).get
-    val accountNumber: String = submission.bankDetails.map(_.accountNumber.get).get
-    val sortCode: String = submission.bankDetails.map(_.sortCode.get).get
-    val installment: String = submission.schedule.map(_.schedule.instalments).getClass.toString
-    val interestTotal: BigDecimal = submission.schedule.map(_.schedule.totalInterestCharged).get
-    val total: BigDecimal = submission.schedule.map(_.schedule.totalPayable).get
+
+    val bankDetails =
+      submission.bankDetails.getOrElse(throw new RuntimeException(s"bank details missing on submission $submission"))
```

Comment:
Could you please move or reuse existing methods to `Journey` class and follow the pattern.
 
See the `Journey` implementation: 
```
...
  def amount: BigDecimal = maybeAmount.getOrElse(throw new RuntimeException(s"Expected 'amount' to be there but was not found. [${_id}] [${this}]"))
  def taxpayer: Taxpayer = maybeTaxpayer.getOrElse(throw new RuntimeException(s"Expected 'Taxpayer' to be there but was not found. [${_id}] [${this}]"))
  def calculatorInput: CalculatorInput = maybeCalculatorData.getOrElse(throw new RuntimeException(s"Expected 'CalculatorData' to be there but was not found. [${_id}] [${this}]"))

```

PS. Could you rename the `submission` to `journey` plase.

Reply:
@pawelpanasewicz this is another of those pieces of string with this service.
You pull it and everything starts to unravel :-)
Andy has merged the branch - will follow up with another pr.